.github/workflows/build.yaml

```diff
@@ -11,7 +11,9 @@ jobs:
     name: Build
     strategy:
       matrix:
-        python-version: [ '3.9', '3.10', 'pypy3.9' ]
+        python-version: ${{ fromJSON(vars.ARCALOT_PYTHON_SUPPORTED_VERSIONS) }}
+        exclude:
+          - python-version: 3.12
```

Comment:
We probably don't want this PR merged with this code.  Instead, we should either [fix the Python SDK](https://github.com/arcalot/arcaflow-lib-kubernetes-python/actions/runs/9654359931/job/26628436184#step:8:153), or remove `3.12` from `ARCALOT_PYTHON_SUPPORTED_VERSIONS`.

Thoughts, @jaredoconnell, @mfleader, @dbutenhof?

Reply:
This was fixed in https://github.com/arcalot/arcaflow-plugin-sdk-python/pull/135; once that is released and the dependency here is updated, this PR can proceed without the above code.